compiler/resolution/implementForallIntents.cpp

```diff
@@ -717,12 +717,15 @@ std::map<ForallStmt*, std::set<Symbol*>> refMaybeConstForallPairs;
 // It is done on an already-existing, explicit shadow variable
 // or before an implicit shadow variable is to be created.
 //
-static void resolveShadowVarTypeIntent(ForallStmt* fs,
+// Returns true if the resolved shadow var intent was an implicit ref-maybe-const
+//
+static bool resolveShadowVarTypeIntent(ForallStmt* fs,
                                        Symbol* sym,
                                        Type*& type,
                                        ForallIntentTag& intent,
                                        bool& prune)
 {
+  bool ret = false;
```

Comment:
This function seems to be "returning" things in `type`, `intent` and `prune` through arguments. Could you "return" the ref-maybe-const-ness of the implicit intent in a similar fashion for consistency?